cypress/integration/movie_list_spec.js

```diff
@@ -0,0 +1,119 @@
+describe('Movie List', function() {
+  beforeEach(function() {
+    cy.visit('/')
+  })
+
+  context('12 Angry Men', function() {
+    beforeEach(function() {
+      cy.get('#movie-list .movie-row:first').as('firstMovie')
+    })
+
+    it('should be first', function() {
+      cy.get('@firstMovie').contains('12 Angry Men')
+    })
+
+    it('should have a title image', function() {
+      cy
+        .get('@firstMovie')
+        .find('.image img')
+        .should('have.attr', 'title', '12 Angry Men')
+    })
+
+    it('should have a linked title heading', function() {
+      cy
+        .get('@firstMovie')
+        .find('.title')
+        .contains('12 Angry Men')
+        .should('match', 'a')
+    })
+
+    it('should have movie info', function() {
+      cy.get('@firstMovie').find('.movie-info')
+    })
+
+    it('should have actors', function() {
+      cy.get('@firstMovie').find('.actors')
+    })
+
+    it('should have the plot', function() {
+      cy.get('@firstMovie').find('.plot')
+    })
+
+    describe('delete button', function() {
+      beforeEach(function() {
+        cy.get('@firstMovie').within(function() {
+          cy.get('.title .glyphicon-remove').click()
+        })
+
+        cy.get('[role=dialog]').as('dialog')
+      })
+
+      it('shows dialog', function() {
+        cy.get('@dialog').contains('Are you sure?')
+      })
+
+      it('does nothing when cancelled', function() {
+        cy
+          .get('@dialog')
+          .contains('Cancel')
+          .click()
+
+        cy.get('@firstMovie').contains('12 Angry Men')
+      })
+
+      it('removes 12 Angry Men when confirmed', function() {
+        cy
+          .get('@dialog')
+          .contains('Ok')
+          .click()
+
+        cy.get('@firstMovie').should('not.contain', '12 Angry Men')
+
+        // @TODO Move to a separate test
+        cy.root().contains('Movie was deleted')
+
+        // Dialog dismissal has an animation
+        cy.wait(1000)
+      })
+    })
+
+    describe('edit button', function() {
+      beforeEach(function() {
+        cy.get('@firstMovie').within(function() {
+          cy.get('.title .glyphicon-edit').click()
+        })
+
+        cy.get('[role=dialog]').as('dialog')
+      })
+
+      it('shows dialog', function() {
+        cy.get('@dialog').contains('Edit movie')
+      })
+
+      it('does nothing when cancelled', function() {
+        cy
+          .get('@dialog')
+          .contains('Cancel')
+          .click()
+
+        cy.get('@firstMovie').contains('12 Angry Men')
+      })
+
+      it('changes the title when edited', function() {
+        cy.get('@dialog').within(function() {
+          cy
+            .get('input[name=title]')
+            .clear()
+            .type('12 Angry Men and a Baby')
+
+          cy.contains('Save').click()
+        })
+
+        cy.get('@firstMovie').contains('12 Angry Men and a Baby')
+
+        // @TODO Move to a separate test
+        cy.root().contains('Movie was saved')
```

Comment:
Toast notifications are tricky.  My preference would be that they would be tested separately (read: stories or unit tests), as they're not _crucial_ to functionality, but a level of polish that makes `cy.screenshot()` difficult to time.